client/landing/jetpack-cloud/index.js

```diff
@@ -43,20 +43,13 @@ export default function() {
 			clientRender
 		);
 		page(
-			'/backups/:site/download',
```

Comment:
this route doesn't really work without ID unless we have some sort of backup picker, so let's just not worry about it for now